mantidproject/mantid
# Add Fit Script Generator to the interface menu

**Description of work.**
This PR adds the Fit Script Generator interface to the interface menu in the `General` category.

**To test:**
1. `Interfaces`->`General`->`Fit Script Generator`
2. Load the data files below into the workbench
3. Click `Add domains`  to add these four domains into the FSG. Then close the Add Domains dialog box
4. Change the End X's to be 20.0
5. Change the Fitting Mode to `Simultaneous` (bottom right)
6. Look at the top right and make sure `All Domains` is selected.
7. Right click the area under the combo box and add a `GausOsc` function.
8. Change `Frequency` to have value `1.3`, and make it Global
9. Now click `Generate Script to Clipboard`
10. Open a new python script window on Workbench, and make sure it is empty. Right click and paste the generated code.
11. Run this python script and wait. You should get four plots representing a simultaneous fit:

![image](https://user-images.githubusercontent.com/40830825/120819809-d7e10700-c54b-11eb-8b41-e584e97868cf.png)

See PR #31719 for testing instructions, except open the interface from the interface menu and load the MUSR data into workbench.
You can also try doing a sequential fit.

**Data Files**
[MUSR62260.zip](https://github.com/mantidproject/mantid/files/8471782/MUSR62260.zip)



Fixes #32331

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
